integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/AndWithAddPredicateBaseIT.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.search.predicate;
+
+import org.hibernate.search.engine.search.predicate.dsl.PredicateFinalStep;
+import org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.BulkIndexer;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.StubMappedIndex;
+import org.hibernate.search.util.impl.test.runner.nested.Nested;
+import org.hibernate.search.util.impl.test.runner.nested.NestedRunner;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(NestedRunner.class)
+public class AndWithAddPredicateBaseIT {
+
+	@ClassRule
+	public static SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	@BeforeClass
+	public static void setup() {
+		setupHelper.start()
+				.withIndexes(
+						ScoreIT.index
+				)
+				.setup();
+
+		final BulkIndexer scoreIndexer = ScoreIT.index.bulkIndexer();
+		ScoreIT.dataSet.contribute( scoreIndexer );
+
+		scoreIndexer.join();
+	}
+
+	@Test
+	public void takariCpSuiteWorkaround() {
+		// Workaround to get Takari-CPSuite to run this test.
+	}
+
+	@Nested
+	public static class ScoreIT extends AbstractPredicateScoreIT {
```

Comment:
Note you could move this test to `AndPredicateBaseIT`: just add another nested test next to `ScoreIT` called `AddScoreIT`, and make sure to initialize its index in `setup()`.

Reply:
Okay!  Not a big step towards less code, but a bigger one towards less files with dumb names!  ;-)